### PR TITLE
make verify compatiable

### PR DESCRIPTION
**Pull request by Rebal67**

### CODE REVIEW

File: src/JWT.ts [hunk 1]
```diff
@@ -165,6 +165,11 @@ export const SUPPORTED_PUBLIC_KEY_TYPES: PublicKeyTypes = {
      *   not an ethereumAddress
      */
     'EcdsaPublicKeySecp256k1',
+    /**
+     * @deprecated, supported for backward compatibility. Equivalent to EcdsaSecp256k1VerificationKey2019 when key is
+     *   not an ethereumAddress
+     */
+    'VerifiableCondition',
```
Comment (theblockstalk) [inline src/JWT.ts]: also need to add a TODO that we need to support the R1 key as well

File: src/JWT.ts [hunk 2]
```diff
@@ -165,6 +165,11 @@ export const SUPPORTED_PUBLIC_KEY_TYPES: PublicKeyTypes = {
      *   not an ethereumAddress
      */
     'EcdsaPublicKeySecp256k1',
+    /**
+     * @deprecated, supported for backward compatibility. Equivalent to EcdsaSecp256k1VerificationKey2019 when key is
```
Comment (theblockstalk) [inline src/JWT.ts]: comments here are not correct. please reference it to the verifiable conditions verification method type

File: src/VerifierAlgorithm.ts [hunk 3]
```diff
@@ -36,6 +36,15 @@ function extractPublicKeyBytes(pk: VerificationMethod): Uint8Array {
     return base64ToBytes((<LegacyVerificationMethod>pk).publicKeyBase64)
   } else if (pk.publicKeyHex) {
     return hexToBytes(pk.publicKeyHex)
+  } else if (pk.conditionWeightedThreshold && Array.isArray(pk.conditionWeightedThreshold)) {
+    return hexToBytes(
```
Comment (theblockstalk) [inline src/VerifierAlgorithm.ts:41]: nothin' to do here yet, but this needs to be recursive as the inner object may be a jwk, another type of verification method or even another verifiable condition

**Review by theblockstalk — COMMENTED** [review]
at some point we want to rename "VerifiableCondition" to something that does not have the VC acronym. Maybe should consider doing this now. Lets talk about it.

File: src/JWT.ts (hunk 1 above)
Comment (Rebal67) [reply src/JWT.ts]: added

File: src/JWT.ts (hunk 2 above)
Comment (Rebal67) [reply src/JWT.ts]: removed comments

File: src/VerifierAlgorithm.ts (hunk 3 above)
Comment (Rebal67) [reply src/VerifierAlgorithm.ts:41]: added

**Pull request merged**